server.go

```diff
@@ -55,7 +63,13 @@ type ServeConfig struct {
 	TLSProvider func() (*tls.Config, error)
 
 	// Plugins are the plugins that are served.
-	Plugins map[string]Plugin
+	// The implied version of this PluginSet is the Handshake.ProtocolVersion.
+	Plugins PluginSet
```

Comment:
Does making this a named type break an existing caller that is just passing directly a `map[string]Plugin`?

Reply:
Nope, the underlying type is the same so they are assignable, and I think this is easier to read than `map[string]map[string]Plugin`.

Reply:
Totally agreed :grinning: ... just wanted to make sure since I recall running into issues with stuff like this in our recent refactoring, but I guess the rules end up a little different for names around primitives than for others.